File: python3/LeetCode/canJump.py
from typing import List


class Solution:
    def canJump(self, nums:List[int]) -> bool:
        end = 0
        for i in range(0, len(nums)):
            if i>end:
                return False

            end = max(end, i+nums[i])
        return True

if __name__ == '__main__':
    nums = {2,3,1,1,4}
    print(Solution().canJump(nums))

File: go/LeetCode/canJump.go
package main

func main() {
	
}

func canJump(nums []int) bool {
	max :=0
	for i:=0;i<len(nums);i++ {
		if i>max {
			return false
		}
		max = max111(max, i+nums[i])
	}
	return true
}

func max111(x int, y int) int {
	if x>y {
		return x
	}else {
		return y
	}
}